sqlmesh/core/reference.py

```
@@ -53,18 +53,66 @@ def name(self) -> str:
 
 class ReferenceGraph:
     def __init__(self, models: t.Iterable[Model]):
-        self._model_refs: t.DefaultDict[str, t.Dict[str, Reference]] = defaultdict(dict)
-        self._ref_models: t.DefaultDict[str, t.Dict[str, str]] = defaultdict(dict)
+        self._model_refs: t.Dict[str, t.Dict[str, Reference]] = {}
+        self._ref_models: t.Dict[str, t.Dict[str, str]] = {}
+        self._dim_models: t.Dict[str, t.Set[str]] = {}
 
         for model in models:
             self.add_model(model)
 
     def add_model(self, model: Model) -> None:
+        """Add a model and its references to the graph.
+
+        Args:
+            model: the model to add.
+        """
+        for column in model.columns_to_types or {}:
+            self._dim_models.setdefault(column, set())
+            self._dim_models[column].add(model.name)
+
         for ref in model.all_references:
+            self._model_refs.setdefault(model.name, {})
             self._model_refs[model.name][ref.name] = ref
+            self._ref_models.setdefault(ref.name, {})
             self._ref_models[ref.name][model.name] = model.name
 
+    def models_for_column(self, source: str, column: str, max_depth: int = 3) -> t.List[str]:
+        """Find all the models with a column that join to a source within max_depth.
+
+        Args:
+            source: The source model.
+            column: The column to look for.
+            max_depth: The maximum number of models to join to find a path.
+
+        Returns:
+            The list of models that fit the criteria of the search.
+        """
+        models = []
+
+        for model in self._dim_models[column]:
+            try:
+                if model != source:
+                    self.find_path(source, model)
+                models.append(model)
+            except SQLMeshError:
+                pass
```

why ignore the error?

---

not every model can join to every other